GitUI/CommandsDialogs/RevisionFileTreeControl.cs

```diff
@@ -971,6 +979,40 @@ public bool SelectFileOrFolder(string filePath)
             return _revisionFileTreeController.SelectFileOrFolder(tvGitTree, filePath[Module.WorkingDir.Length..]);
         }
 
+        private class ScriptOptionsProvider : IScriptOptionsProvider
+        {
+            private RevisionFileTreeControl _revisionFileTreeControl;
+
+            public ScriptOptionsProvider(RevisionFileTreeControl revisionFileTreeControl)
+            {
+                _revisionFileTreeControl = revisionFileTreeControl;
+            }
+
+            IReadOnlyList<string> IScriptOptionsProvider.Options => new[] { "SelectedFiles", "LineNumber" };
```

Comment:
Use consts.

Also, the same literals are used in multiple places, which is undesirable. Please defined the consts in one place and reuse them, where required.

Reply:
Where should such a constant be defined?
Also, doesn't it kinda defeat the "pluggable options" design, if the options are still centralized?

Reply:
> Where should such a constant be defined?

Refer to my generic implementation of `ScriptOptionsProvider` - which might be renamed to `FileScriptOptionsProvider` or the like.